assets/js/app.js

```diff
@@ -72,7 +72,344 @@ Hooks.Logs = {
   }
 }
 
+function generateEchartObjects(heading , echartContainer ,  x_axis_data){
```

Comment:
You should extract all of this logic into a dedicated file, so you can import it into the app.js

Reply:
noted